src/integrations/prefect-dbt/tests/cli/test_commands.py

```diff
@@ -513,6 +685,31 @@ async def test_flow():
     assert "Unsuccessful Nodes" in a.data
 
 
+@pytest.mark.usefixtures("dbt_runner_freshness_error")
+async def test_run_dbt_model_creates_unsuccessful_artifact_for_source_command(
+    profiles_dir, dbt_cli_profile_bare
+):
+    @flow
+    async def test_flow():
+        return trigger_dbt_cli_command(
+            command="dbt source freshness",
+            profiles_dir=profiles_dir,
+            dbt_cli_profile=dbt_cli_profile_bare,
+            summary_artifact_key="foo",
+            create_summary_artifact=True,
+        )
+
+    with pytest.raises(
+        Exception, match="dbt task result success: False with exception: None"
+    ):
+        await test_flow()
+    assert (a := await Artifact.get(key="foo"))
+    assert a.type == "markdown"
+    assert a.data.startswith("# dbt source freshness Task Summary")
+    assert "my_first_dbt_model" in a.data
+    assert "Unsuccessful Nodes" in a.data
```

Comment:
related to the comment above, can you explain how this aligns with the exclusion logic for source nodes?

Reply:
Main point of this test is to just confirm that unsuccessful runs of `dbt source freshness` generate the summary artifact, which before the fix did not.